app/src/main/java/com/github/wnder/GuestUser.java

```diff
@@ -11,4 +18,31 @@ public String getName(){
     public Uri getProfilePicture(){
         return Uri.parse("android.resource://com.github.wnder/" + R.raw.ladiag);
     }
+
+    /**
+     * Returns the id of a picture existing in the db
+     * @return the id of the picture, an empty string if non is eligible
+     * @throws ExecutionException
+     * @throws InterruptedException
+     */
+    public String getNewPicture() throws ExecutionException, InterruptedException{
+        Storage storage = new Storage();
+
+        //Get the ids of all the uploaded pictures
+        CompletableFuture<Set<String>> allIdsFuture = storage.getIdsOfAllUploadedPictures();
+        Set<String> allIds = allIdsFuture.get();
+
+        //If no image fits, return empty string
+        if(0 == allIds.size()){
+            return new String();
+        }
+        //else, return randomly chosen string
+        else{
+            List<String> idList = new ArrayList<>();
+            idList.addAll(allIds);
+            Random random = new Random();
+            int index = random.nextInt(allIds.size());
```

Comment:
How do you make sure this is a new photo that the user never got before ? I know this is for a guest but is there no  possibility to use a cache ?